ripser/ripser.py

```diff
@@ -37,6 +37,82 @@
 from pyRipser import doRipsFiltrationDMSparse as DRFDMSparse
 
 
+def get_greedy_perm_pc(X, N = -1, metric="euclidean"):
+    """
+    Compute a furthest point sampling permutation of a set
+    of points
+    Parameters
+    ----------
+    X: ndarray(N, d)
+        Point cloud with N points in d dimensions
+    N: int
+        Number of points to take in the permutation
+    metric: string or callable
+        The metric to use when calculating distance between instances in a 
+        feature array. If metric is a string, it must be one of the options 
+        specified in PAIRED_DISTANCES, including "euclidean", "manhattan", 
+        or "cosine". Alternatively, if metric is a callable function, it is 
+        called on each pair of instances (rows) and the resulting value 
+        recorded. The callable should take two arrays from X as input and 
+        return a value indicating the distance between them.
+    Returns
+    -------
+    perm: ndarray(N)
+        Indices of points in the greedy permutation
+    lambdas: ndarray(N)
+        Covering radii at different points
+    """
+    if N == -1:
```

Comment:
If `N=None`, then you could do `N = N or X.shape[0]`. Can we assume 0 is not a valid input?